source/extensions/filters/network/mysql_proxy/mysql_codec.h

```diff
@@ -109,18 +109,15 @@ class MySQLCodec : public Logger::Loggable<Logger::Id::filter> {
 
   virtual ~MySQLCodec() {}
 
-  int decode(Buffer::Instance& data, uint64_t& offset, int seq, int len) {
+  int decode(Buffer::Instance& data, int seq, int len) {
     seq_ = seq;
-    const uint64_t prev_offset = offset;
-    int result = parseMessage(data, offset, len);
-    offset = prev_offset + len; // Ensure that the whole message was consumed
-    return result;
+    return parseMessage(data, len);
   }
 
   virtual std::string encode() PURE;
 
 protected:
-  virtual int parseMessage(Buffer::Instance& data, uint64_t& offset, int len) PURE;
+  virtual int parseMessage(Buffer::Instance& data, int len) PURE;
```

Comment:
Not in this change but can you update `int` to a proper unsigned type? Same above for `decode` and other places it applies in this PR? Thank you.

Reply:
Addressed in 9968c6a.